src/concurrency/local_epoch.cpp

```diff
@@ -16,61 +16,53 @@
 namespace peloton {
 namespace concurrency {
 
-  bool LocalEpoch::EnterEpoch(const uint64_t epoch_id) {
-    epoch_lock_.Lock();
-    // if this thread is never used or GC'd
-    if (epoch_lower_bound_ == UINT64_MAX) {
-      epoch_lower_bound_ = epoch_id - 1;
-    } else if (epoch_lower_bound_ >= epoch_id) {
-    // epoch_lower_bound_ has already been updated by the GC.
-    // have to grab a newer epoch_id.
-      epoch_lock_.Unlock();
-      return false;
-    } 
+  bool LocalEpoch::EnterEpoch(const eid_t epoch_id, const TimestampType ts_type) {
 
-    auto epoch_map_itr = epoch_map_.find(epoch_id);
-    // check whether the corresponding epoch exists.
-    if (epoch_map_itr == epoch_map_.end()) {
+    epoch_lock_.Lock();
 
-      std::shared_ptr<Epoch> epoch_ptr(new Epoch(epoch_id, 1));
+    // if this thread is never used or has been GC'd
+    if (epoch_id_lower_bound_ == UINT64_MAX) {
 
-      epoch_queue_.push(epoch_ptr);
-      epoch_map_[epoch_id] = epoch_ptr;
+      epoch_id_lower_bound_ = epoch_id - 1;
+    
+    } else if (epoch_id_lower_bound_ >= epoch_id) {
```

Comment:
The local epoch progress can be faster than the global one. In this case, `epoch_id_lower_bound_` can be larger than `epoch_id`.